src/App.tsx

```diff
@@ -1,35 +1,181 @@
+            ? conn.recipient.accountId
+            : conn.requester.accountId,
+        username:
+          conn.recipient.accountId !== userId
+            ? conn.recipient.username
+            : conn.requester.username,
+      }));
+      setMyConnections(mappedConnections);
+    } catch (error) {
+      console.error('Error fetching connections:', error);
+    }
+  };
+
+  const handleSendConnectionRequest = async (recipientId: number) => {
+    try {
+      if (recipientId !== userId) {
+        await sendConnectionRequest(userId, recipientId);
+        toast.success(`You have sent a connection request to ${searchResults.find(user => user.id === recipientId)?.username}!`);
+        await fetchPendingRequests();
+      } else {
+        toast.error("You cannot send a connection request to yourself.");
+      }
+    } catch (error) {
+      toast.error('Error sending connection request.');
+      console.error('Error sending connection request:', error);
+    }
+  };
+
+  const handleAcceptRequest = async (requestId?: number) => {
+    if (requestId) {
+      try {
+        await acceptConnectionRequest(requestId);
+        toast.success('Connection request accepted!');
+        await fetchPendingRequests();
+        await fetchMyConnections();
+      } catch (error) {
+        toast.error('Error accepting request.');
+        console.error('Error accepting request:', error);
+      }
+    }
+  };
+
+  const handleRejectRequest = async (requestId?: number) => {
+    if (requestId) {
+      try {
+        await rejectConnectionRequest(requestId);
+        toast.info('Connection request rejected.');
+        await fetchPendingRequests();
+      } catch (error) {
+        toast.error('Error rejecting request.');
+        console.error('Error rejecting request:', error);
+      }
+    }
+  };
+
+  const handleRemoveConnection = async (connectionId?: number) => {
+    if (connectionId) {
+      try {
+        await removeConnection(connectionId);
+        toast.info('Connection removed.');
+        await fetchMyConnections();
+      } catch (error) {
+        toast.error('Error removing connection.');
+        console.error('Error removing connection:', error);
+      }
+    }
+  };
 
   return (
-    <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
-}
+    <div className="dashboard">
```

Comment:
We may want to leave `App.tsx` to bootstrap our app, not necessarily create layouts in it. We can extract these components and `App.tsx` can render our app's routes, error boundaries, contexts, etc.

Reply:
I will make sure App.tsx is clean